openwisp_monitoring/check/classes/base.py

```diff
@@ -0,0 +1,48 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from swapper import load_model
+
+from openwisp_controller.config.models import Device
+
+Check = load_model('check', 'Check')
+Metric = load_model('monitoring', 'Metric')
+
+
+class BaseCheck(object):
```

Comment:
While refactoring BaseCheck is needed, this kind of changes make it more difficult to review PRs and detect problems later. It is usually a good thing to make one change per PR so please consider it next time (1 PR for refactoring and another one to add this new check).